### PR TITLE
docs: ADR-001 remove default redaction of IP addresses

### PR DESCRIPTION
**Pull request by xavpaice**

### CODE REVIEW

File: docs/arch/adr-001-ipaddress-redaction.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  into the product in addition to the defined spec is confusing and, in some cases, problematic if you do not want
+  the built in redactors.
+* A default set of redactors gives product users a false sense of security in that though the default redactors
+  do cover some redaction, they do not cover every possible combination for every particular sensitive item.
+  Redaction of hostnames is almost impossible given the freeform nature of hostnames, and the built in Password
+  redactor does not cover all passwords, merely a particular json combination.
+
+## Decision
+
+We will remove the IP address redactor from Troubleshoot.
```
Comment (banjoh) [inline docs/arch/adr-001-ipaddress-redaction.md]: I'm guessing the wording here needs to include "making it optional"

File: docs/arch/adr-001-ipaddress-redaction.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+## Solution
+
+The following changes need to be made:
+
+* [Documentation](https://troubleshoot.sh/docs/redact/ip-addresses/) for the IP address redaction needs altering to reflect that Troubleshoot
+  does not automatically redact IP addresses, but if users wish to there is an example yaml spec available.
+* Removal of the IP address redactor from [the code](https://github.com/replicatedhq/troubleshoot/blob/v0.45.0/pkg/redact/redact.go#L170)
+* Clear release notes on release of this change, communicating that those wishing to redact IP addresses need to add that redactor to their code.
```
Comment (banjoh) [inline docs/arch/adr-001-ipaddress-redaction.md]: ...those wishing to redact IP addresses need to add that redactor to their code **or spec**.

File: docs/arch/readme.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+* Think of the document as a conversation with a future developer. This means write well and use full
+  sentences.
+* Update consequences as they become known.  The ADR becomes like a diary for seeing how the design
+  decisions we make impact the system over time.
+* Include diagrams as necessary.
+
+
+## References
+
+Nygard, Michael. Documenting Architecture Decisions, from _Think Relevance_ blog. [Web](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions)
```
Comment (banjoh) [inline docs/arch/readme.md]: I think https://adr.github.io/ as a reference is good to have

File: docs/arch/readme.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+If superseded, include a link to the new ADR
+
+
+## Consequences
+
+Describe the resulting context, after applying the decision. All consequences should be listed here,
+not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences,
+but all of them affect the team and project in the future.
+```
```
Comment (banjoh) [inline docs/arch/readme.md]: I would add [this ADR tools](https://adr.github.io/#decision-capturing-tools) link for people who would prefer to generate the ADR using the CLI.

**Review by banjoh — DISMISSED** [review]

**Comment by xavpaice** [comment]
needs a proper rebase

File: docs/arch/adr-001-ipaddress-redaction.md (hunk 1 above)
Comment (xavpaice) [reply docs/arch/adr-001-ipaddress-redaction.md]: it's removing it from the list of hard coded default redactors - yeah it's removing, but optionally can be added also.  Will re-word.

**Review by diamonwiggins — DISMISSED** [review]
lgtm

File: docs/arch/readme.md (hunk 3 above)
Comment (xavpaice) [reply docs/arch/readme.md]: agreed - separate commit though if you could add it at some point?

File: docs/arch/readme.md (hunk 4 above)
Comment (xavpaice) [reply docs/arch/readme.md]: agreed - separate commit though if you could add it at some point?

**Comment by xavpaice** [comment]
> needs a proper rebase

done

File: docs/arch/adr-001-ipaddress-redaction.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+in 2022 (as of October 2022). This doesn't cover uploaded files, phone calls, screen shares, etc. It also
+doesn't include anything else networking like mac address, hostnames, ports.
+
+The problem discussed in this document:
+
+* Users of Troubleshoot need to avoid redaction if they have networking issues, in order to share IP addresses
+  that are redacted by default.
+* The user experience of the product is based around "define what you want", however with default redactors built
+  into the product in addition to the defined spec is confusing and, in some cases, problematic if you do not want
+  the built in redactors.
```
Comment (chris-sanders) [inline docs/arch/adr-001-ipaddress-redaction.md:27]: I really like this point, I wonder is this the correct place to also consider the intention to move to a fully "define what you want" model? Maybe in the decision section below we can state that the intention is to review other redactors with the expectation of eventually removing all 'default' redaction?

**Review by chris-sanders — APPROVED** [review]
Just one comment, I'm going to approve because I'm fine if you decide to make the suggested change or not. I think both are valid decisions and this is a good ADR either way.

File: docs/arch/adr-001-ipaddress-redaction.md (hunk 2 above)
Comment (chris-sanders) [reply docs/arch/adr-001-ipaddress-redaction.md]: Should this just say "to their spec"? What "code" would we be referencing here?

File: docs/arch/adr-001-ipaddress-redaction.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+## Solution
+
+The following changes need to be made:
+
+* [Documentation](https://troubleshoot.sh/docs/redact/ip-addresses/) for the IP address redaction needs altering to reflect that Troubleshoot
+  does not automatically redact IP addresses, but if users wish to there is an example yaml spec available.
+* Removal of the IP address redactor from [the code](https://github.com/replicatedhq/troubleshoot/blob/v0.45.0/pkg/redact/redact.go#L170)
+* Clear release notes on release of this change, communicating that those wishing to redact IP addresses need to add that redactor to their spec.
+* Possible broadcast communication since this is a change to default behavior.
```
Comment (DexterYan) [inline docs/arch/adr-001-ipaddress-redaction.md:46]: will it be considerate as a possible broadcast communication way? Like adding a warning message when running the previous troubleshoot cli. for example
```
The IP address redaction usage was deprecated in Troubleshoot v0.* and will be removed in Troubleshoot v0.*+1.
```

**Pull request merged**